src/components/dashboard-widgets/dashboard-text-widget/dashboard-text-widget.spec.ts

```diff
@@ -0,0 +1,66 @@
+
+@Component({
+    selector: 'ux-text-widget-test-component',
+    template: `
+        <ux-dashboard>
+            <ux-dashboard-text-widget id="widget-text" name="Text Widget" heading="Text Widget"
+                                      fixedMode="false" colSpan="3" rowSpan="4"
+                                      [text]="loremIpsum" editable="true">
+            </ux-dashboard-text-widget>
+        </ux-dashboard>
+    `
+})
+export class DashboardTextWidgetTestComponent {
+    loremIpsum: string = loremIpsum;
+    @ViewChild(DashboardTextWidgetComponent) widget: DashboardTextWidgetComponent;
+}
+
+describe('Text Widget', () => {
+    let component: DashboardTextWidgetTestComponent;
+    let fixture: ComponentFixture<DashboardTextWidgetTestComponent>;
+    let nativeElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [DashboardModule, DashboardPredefinedWidgetsModule],
+            declarations: [DashboardTextWidgetTestComponent]
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(DashboardTextWidgetTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    });
+
+    it('should initialise correctly', () => {
+        expect(component).toBeTruthy();
+        expect(component.widget).toBeTruthy();
+    });
+
+    it('should set inputs correctly', () => {
+        expect(component.widget.fixedMode).toEqual(false);
+        expect(component.widget.colSpan).toEqual(3);
+        expect(component.widget.rowSpan).toEqual(4);
+        expect(component.widget.id).toEqual('widget-text');
+        expect(component.widget.name).toEqual('Text Widget');
+        expect(component.widget.heading).toEqual('Text Widget');
+
+        expect(component.widget.text).toEqual(loremIpsum);
+        expect(component.widget.editable).toEqual(true);
+        expect(component.widget.editLabel).toEqual('Edit');
+        expect(component.widget.showFullTextLabel).toEqual('Show full text');
+        expect(component.widget.saveLabel).toEqual('Save');
+        expect(component.widget.cancelLabel).toEqual('Cancel');
+        expect(component.widget.closeLabel).toEqual('Close Side Panel');
+    });
```

Comment:
This goes for all spec files for each of the widgets. We don't need to test that inputs are being set, this is just testing the framework. These can be removed.

I think we would need more test coverage on these testing actual functionality for each widget. For example, saving the text on this one.